doc/sphinx/proof-engine/ltac.rst

```diff
@@ -1124,7 +1126,8 @@ Pattern matching on terms: match
       | multimatch
       match_pattern ::= @cpattern
       | context {? @ident } [ @cpattern ]
-      cpattern ::= @term
+      cpattern ::= {? Qed } @one_term   SSR
```

Comment:
This is a hack, IIRC we add a non-terminal using the nice macros and later we extend it with the low level syntax  adding the real parsing rule. Since the nice macros do not let you declare a non-terminal without also declaring a parsing rule, we declare a broken parsing rules (which should never parse). I think here `"   Qed   "` makes it unparsable because of the spaces.

Reply:
I can hide the production, but there are no spaces around "Qed" (bottom).

Indeed, it is accepted:

```
From Coq Require Import ssreflect ssrfun ssrbool.
Goal True.
set x := { 1 } Qed True.

1 subgoal
x := True : Prop
______________________________________(1/1)
x
```

```
ARGUMENT EXTEND lcpattern
     TYPED AS cpattern
     PRINTED BY { pr_ssrterm }
     INTERPRETED BY { interp_ssrterm }
     GLOBALIZED BY { glob_cpattern } SUBSTITUTED BY { subst_ssrterm }
     RAW_PRINTED BY { pr_ssrterm }
     GLOB_PRINTED BY { pr_ssrterm }
| [ "Qed" lconstr(c) ] -> { mk_lterm c None }
END
```

Reply:
@maximedenes you had a patch to make grammar rules optionals in `ARGUMENT EXTEND`. Did it got merged?

In the meanwhile please hide it.

Reply:
I see this was done for both `lcpattern` and `cpattern`.  I will hide both.  But you may want to change it to " Qed " so the production won't be used.